src/core/actions/actionState.ts

```diff
@@ -0,0 +1,67 @@
+/**
+ * @license
+ * SKALE Metaport
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+/**
+ * @file actionState.ts
+ * @copyright SKALE Labs 2023-Present
+ */
+
+export type ActionState =
+    | 'init'
+    | 'approve'
+    | 'approveDone'
+    | 'transfer'
+    | 'transferDone'
+    | 'received'
+    | 'transferETH'
+    | 'transferETHDone'
+    | 'receivedETH'
```

Comment:
Is this only for M2S ETH or is this for all native token transfers i.e does the sFUEL to wSFUEL use this path? If yes, suggest it is changed to transferNative over transferETH to encompass all native asset movement

Reply:
@TheGreatAxios it's only for ETH M2S and S2M, not for sFUEL